### DISCUSSION TITLE
Link camera position to objects instead of fixed coordinates

### DISCUSSION THREAD

**Discussion by andreas65m** [senshu/Sozi#681] [Ideas]
Hello,

I created a big graph using graphviz and would like to present it using sozi.
Every time I add a node ( can happen until the beginning of the presentation ) to the graph, graphviz moves other nodes. 
This way I have to created the sozi-video from scratch.
Would it be possible to store the names of the nodes as alternative to the fixed coordinates in sozi?
This way the camera would always move to the right point.

Greetings
Andreas

**Comment by aumouvantsillage** [comment]
> Would it be possible to store the names of the nodes as alternative to the fixed coordinates in sozi?

This is already what Sozi does.
When you create a frame, Sozi chooses an SVG element that will serve as an "anchor" in each layer.
Sozi also stores absolute coordinates but they are used only if the anchor element is not found.

This technique works only if the SVG elements have IDs, and if these IDs are preserved when the document is modified.
There are tools that generate SVG documents with no IDs, and there are tools that generate a different set of IDs each time the SVG is saved or exported.
Maybe Graphviz falls into one of these categories.

**Comment by andreas65m** [comment]
Salut  Guillaume,
thank you very much for the fast answer. 
Knowing what to search for I discovered that it is possible to set fixed IDs in graphviz

Merci beaucoup
Andreas
